pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ClientCachedDao.java

```diff
@@ -0,0 +1,119 @@
+package org.broadinstitute.ddp.db.dao;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+
+import javax.cache.Cache;
+import javax.cache.expiry.Duration;
+
+import org.broadinstitute.ddp.cache.CacheService;
+import org.broadinstitute.ddp.db.dto.ClientDto;
+import org.broadinstitute.ddp.security.StudyClientConfiguration;
+import org.jdbi.v3.core.Handle;
+
+public class ClientCachedDao extends SQLObjectWrapper<ClientDao> implements ClientDao {
+    private static final String CACHE_NAME = "StudyClientConfigCache.byStudyClientConfigCacheKey";
+
+    private class StudyClientConfigCacheKey implements Serializable {
+        String auth0ClientId;
+        String auth0Domain;
+
+        StudyClientConfigCacheKey(String auth0ClientId, String auth0Domain) {
+            this.auth0ClientId = auth0ClientId;
+            this.auth0Domain = auth0Domain;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            StudyClientConfigCacheKey that = (StudyClientConfigCacheKey) o;
+
+            if (auth0ClientId != null ? !auth0ClientId.equals(that.auth0ClientId) : that.auth0ClientId != null) {
+                return false;
+            }
+            return auth0Domain != null ? auth0Domain.equals(that.auth0Domain) : that.auth0Domain == null;
+        }
+
```

Comment:
In this class at this time, I am only trying to cache the results of:
```java
public StudyClientConfiguration getConfiguration(String auth0ClientId, String auth0Domain)
```
You can see the implementation below.
The tricky thing with caching the result of this method is that it depends on 2 arguments, but cache api takes only one key value.
So what you need to do is to create an object that combines the two arguments.

You also need to implement hashCode and equals to enable HashMap lookups. (Here I just auto-generated it fro IDE).

Also important: everything that goes into cache (keys and values) needs to be ```Serializable```.
Will note that the JCache redisson library does support other methods of serialization of objects, including Jackson JSON serialization, but the Java serialization was easiest to implement. GSON serialization is theoretically possible, but would need to be implemented as extension to Redisson.